app/code/Magento/Store/Model/Store.php

```diff
@@ -666,12 +658,13 @@ public function getBaseUrl($type = UrlInterface::URL_TYPE_LINK, $secure = null)
                     throw new \InvalidArgumentException('Invalid base url type');
             }
 
-            if (false !== strpos($url, self::BASE_URL_PLACEHOLDER)) {
+            if (is_string($url) && strpos($url, self::BASE_URL_PLACEHOLDER) !== false) {
```

Comment:
Should be fixed in tests unless we have a case when `null` is valid config value for installed Magento.

Reply:
@xmav, I fixed code in tests, please check if everything is well.

Thank you.